pkg/reporter/purl/purl.go

```diff
@@ -0,0 +1,49 @@
+package purl
+
+import (
+	"github.com/google/osv-scanner/pkg/models"
+	"github.com/package-url/packageurl-go"
+)
+
+type ParameterExtractor func(packageInfo models.PackageInfo) (namespace string, name string, ok bool)
+
+var ecosystemToPurlMapper = map[models.Ecosystem]string{
+	models.EcosystemMaven:     packageurl.TypeMaven,
+	models.EcosystemGo:        packageurl.TypeGolang,
+	models.EcosystemPackagist: packageurl.TypeComposer,
+	models.EcosystemPyPI:      packageurl.TypePyPi,
+	models.EcosystemRubyGems:  packageurl.TypeGem,
+	models.EcosystemNuGet:     packageurl.TypeNuget,
+	models.EcosystemNPM:       packageurl.TypeNPM,
+}
+
+var ecosystemPURLExtractor = map[models.Ecosystem]ParameterExtractor{
```

Comment:
[nit] you should either use one notation or another `Purl` as above or `PURL` as here..

Reply:
You're right :) Done